src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -568,14 +572,18 @@ void MeshForwarder::HandleDataRequest(const Mac::Address &aMacSource, const otTh
     child->ResetLinkFailures();
     indirectMsgCount = child->GetIndirectMessageCount();
 
+#if OPENTHREAD_CONFIG_ENABLE_VERIFY_ACK_FP_FLAG
+    VerifyOrExit(aFrame.IsAckWithFramePending());
```

Comment:
If we exit, we skip the log below.
How about using this to decide if to set `child->SetDataRequestPending(true)` or not (from within the `if`)?

Reply:
Were there any considerations for posting schedule transmission task even no pending data for SED? I am considering move the log to after `exit:`.

Reply:
I think that would work ok

Reply:
I moved the log before this line because it uses `child` and `indirectMsgCount`.